src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs

```diff
@@ -169,6 +169,12 @@ public void Scan()
 
                     // Tick the heartbeat, if the connection is still active
                     connection.TickHeartbeat();
+
+                    if (connection.AuthenticationExpiration < utcNow && !connection.ConnectionClosedRequested.IsCancellationRequested)
```

Comment:
Why isn't this gated by EnableAuthenticationExpiration

Reply:
It technically is because we don't set the value unless the option is enabled, but if we always set it like you suggest then we'll need to add the check here.

Reply:
```
> null < DateTimeOffset.UtcNow
(1,1): warning CS0464: Comparing with null of type 'DateTimeOffset?' always produces 'false'
false
```

😆

Reply:
Haha, I was using that feature intentionally...